src/pybind/mgr/nfs/export.py

```diff
@@ -424,63 +401,199 @@ def _create_user_key(self, entity, path, fs_name, fs_ro):
             'caps': ['mon', 'allow r', 'osd', osd_cap, 'mds', 'allow {} path={}'.format(
                 access_type, path)],
             'format': 'json',
-            })
+        })
 
         json_res = json.loads(out)
         log.info("Export user created is {}".format(json_res[0]['entity']))
         return json_res[0]['entity'], json_res[0]['key']
 
-    def create_export(self, fs_name, cluster_id, pseudo_path, read_only, path):
+    def create_cephfs_export(self,
+                             fs_name: str,
+                             cluster_id: str,
+                             pseudo_path: str,
+                             read_only: bool,
+                             path: str,
+                             squash: str,
+                             clients: list = []) -> Tuple[int, str, str]:
         if not check_fs(self.mgr, fs_name):
             raise FSNotFound(fs_name)
 
         pseudo_path = self.format_path(pseudo_path)
-        ValidateExport.pseudo_path(pseudo_path)
 
         if cluster_id not in self.exports:
             self.exports[cluster_id] = []
 
-        if not self._fetch_export(pseudo_path):
-            ex_id = self._gen_export_id()
-            user_id = f"{cluster_id}{ex_id}"
-            user_out, key = self._create_user_key(user_id, path, fs_name, read_only)
-            access_type = "RW"
-            if read_only:
+        if not self._fetch_export(cluster_id, pseudo_path):
+            ex_id = self._gen_export_id(cluster_id)
+            user_id = f"nfs.{cluster_id}.{ex_id}"
+            user_out, key = self._create_user_key(
+                cluster_id, user_id, path, fs_name, read_only
+            )
+            if clients:
+                access_type = "none"
+            elif read_only:
                 access_type = "RO"
+            else:
+                access_type = "RW"
             ex_dict = {
-                    'path': self.format_path(path),
-                    'pseudo': pseudo_path,
-                    'cluster_id': cluster_id,
-                    'access_type': access_type,
-                    'fsal': {"name": "CEPH", "user_id": user_id,
-                             "fs_name": fs_name, "sec_label_xattr": ""},
-                    'clients': []
-                    }
+                'path': self.format_path(path),
+                'pseudo': pseudo_path,
+                'cluster_id': cluster_id,
+                'access_type': access_type,
+                'squash': squash,
+                'fsal': {"name": "CEPH", "user_id": user_id,
+                         "fs_name": fs_name, "sec_label_xattr": "",
+                         "cephx_key": key},
+                'clients': clients
+            }
             export = Export.from_dict(ex_id, ex_dict)
-            export.fsal.cephx_key = key
-            self._save_export(export)
+            export.validate(self.mgr)
```

Comment:
There is no need to validate everything. Only if values are not default ones, we should validate. 

https://github.com/ceph/ceph/blob/698eb40db5179cba4ae187054b0dae236154d3e6/src/pybind/mgr/nfs/export_utils.py#L476-L512

Reply:
making validate() a member that validates the entire instance was a deliberate change

Reply:
Most of the values are set by us in this case. Please see my earlier comment